ui/src/components/features/navbar/components/thread-item.tsx

```diff
@@ -13,29 +13,32 @@ import { ChangeEvent, useEffect, useState } from "react";
 
 type TThreadItemProps = {
   thread: TThread;
+  currentThreadId: string | undefined;
 };
 
-export default function ThreadItem({ thread }: TThreadItemProps) {
+export default function ThreadItem({
+  thread,
+  currentThreadId,
+}: TThreadItemProps) {
   const [isEditing, setIsEditing] = useState(false);
   const [editedName, setEditedname] = useState(thread.name || "New thread");
-  const [isSelected, setIsSelected] = useState(false);
   const [isDeleting, setIsDeleting] = useState(false);
   const [isHovering, setIsHovering] = useState(false);
 
-  const {mutate: updateThread, variables: optimisticThread, isPending} = useUpdateThread(thread.id!);
+  const {
+    mutate: updateThread,
+    variables: optimisticThread,
+    isPending,
+  } = useUpdateThread(thread.id!);
   const deleteThread = useDeleteThread(thread.id!);
 
   const router = useRouter();
-  const pathname = usePathname();
   const { toast } = useToast();
   const { assistantId } = useSlugRoutes();
-
-  useEffect(() => {
-    setIsSelected(pathname.includes(thread.id!));
-  }, [pathname, thread.id]);
+  const isCurrentThread = currentThreadId && currentThreadId === thread.id;
```

Comment:
Yeah. I think I need to consider derived state more. It's a bit of a newer concept to me, so I tend to forget about it!